spec/unit/pushprocessor.spec.ts

```diff
@@ -12,21 +22,21 @@ describe("NotificationService", function () {
 
     let pushProcessor: PushProcessor;
 
-    const msc3914RoomCallRule = {
+    const msc3914RoomCallRule: IPushRule = {
         rule_id: ".org.matrix.msc3914.rule.room.call",
         default: true,
         enabled: true,
         conditions: [
             {
-                kind: "event_match",
+                kind: ConditionKind.EventMatch,
```

Comment:
ftr I am generally deeply ambivalent about using symbolic constants in test code: using the literal gives us a chance to check that the constant is doing what we expect.

(This brought to you on the back of a Synapse bug where we had a test which used what we *thought* was a constant but turned out not to be.)

YMMV though, not a particular request to change this.

Reply:
> ftr I am generally deeply ambivalent about using symbolic constants in test code: using the literal gives us a chance to check that the constant is doing what we expect.

Its preferable to `@ts-ignore`.

![image](https://github.com/matrix-org/matrix-js-sdk/assets/2403652/c409923a-c90b-4f61-b96a-2a7ddb036535)

Reply:
I mean I'd argue your problem here is forcing msc3914RoomCallRule to be an IPushRule, but whatever

Reply:
Even if I remove that type cast there, then I'd get the failure in the call to `rewritePushRules`

The only reason it was fine before this PR is because we passed `matrixClient.pushRules!` to `rewritePushRules` and had
![image](https://github.com/matrix-org/matrix-js-sdk/assets/2403652/468587bb-eb31-49fb-9f5e-e94771b14007) to make TS happy with the types of `pushRules` within the mock `matrixClient`

Reply:
![image](https://github.com/matrix-org/matrix-js-sdk/assets/2403652/a316cd41-fdce-4300-a4df-65d5b52be9f7)